rust/agama-server/src/web.rs

```diff
@@ -43,13 +53,29 @@ pub fn service(config: ServiceConfig, events: EventsSender) -> Router {
 ///
 /// * `events`: channel to send the events to.
 pub async fn run_monitor(events: EventsSender) -> Result<(), ServiceError> {
-    let presenter = EventsProgressPresenter::new(events);
+    let presenter = EventsProgressPresenter::new(events.clone());
     let connection = connection().await?;
-    let mut monitor = ProgressMonitor::new(connection).await?;
+    let mut monitor = ProgressMonitor::new(connection.clone()).await?;
     tokio::spawn(async move {
         if let Err(error) = monitor.run(presenter).await {
             eprintln!("Could not monitor the D-Bus server: {}", error);
         }
     });
+    tokio::spawn(run_events_monitor(connection, events.clone()));
+
+    Ok(())
+}
+
+/// Emits the events from the system streams through the events channel.
+///
+/// * `connection`: D-Bus connection.
+/// * `events`: channel to send the events to.
+pub async fn run_events_monitor(dbus: zbus::Connection, events: EventsSender) -> Result<(), Error> {
+    let stream = software_stream(dbus).await?;
```

Comment:
so if I get it, here in the end will be list of all our streams and we will all  mix them, right?

Reply:
Yes, that's the idea. We combine our streams and emit then through the websocket.